### PR TITLE
Feat/cse 265 refactor server

### PR DESCRIPTION
**Pull request by danvanhorn**
# Changes introduced

I refactored the server and routes. Integrated the insertion script, search and filter endpoints. 

## Related issue(s):

- CSE-265

## Contributor checklist

- [x] Wrote/updated tests for introduced changes
- [ ] ~~Added new stories for created/modified components (if applicable)~~
- [ ] Updated Postman library for created/modified routes (if applicable)
- [x] Verified that there are no issues in CircleCI
- [x] Assigned yourself to the PR
- [x] Removed `WIP:` from the PR title
- [X] Requested review from the SB Reviewers team

## Reviewer checklist

- [ ] Assigned yourself to the PR
- [ ] Read through the changes in the `Files changed` tab
- [ ] Verified that tests were written/modified
- [ ] ~~Verified that stories were written/modified (if applicable)~~
- [ ] Verified that Postman was updated (if applicable)
- [ ] Checked out the branch and tested changes locally

### CODE REVIEW

File: src/dal/interface/index.ts [hunk 1]
```diff
@@ -13,20 +13,39 @@ export class DbClient {
...
             throw err;
         }
-        client.close();
+    }
+
+    public async insert(json: object[]) {
+        let result: InsertWriteOpResult;
+        // tslint:disable:no-any
+        let collections: Collection<any>[];
+        if (this.db) {
```
Comment (puregarlic) [inline src/dal/interface/index.ts]: Shouldn't we have a condition here to catch the case when `this.db` doesn't exist?

File: tslint.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "@osu-cass/tslint-config",
   "rules": {
-    "strict-boolean-expressions": [true, "allow-string", "allow-number", "allow-boolean-or-undefined", "allow-mix"]
+    "strict-boolean-expressions": false
```
Comment (puregarlic) [inline tslint.json]: You don't need to do this. Update `@osu-cass/tslint-config` to the latest version, and you'll get this in the configuration without having to override.

File: src/server/index.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    constructor() {
+        this.app = e();
+        this.port = process.env.PORT as string || 3000 as number;
+        this.configure();
+        this.routes();
+        this.client = new DbClient({
+            url: 'mongodb://mongo',
+            port: 27017,
+            dbName: 'admin'
```
Comment (puregarlic) [inline src/server/index.ts]: We should create a new database, specific to Content Specification Explorer. I'm aware that it means we'll need additional logic to check for its existence, but I think it's safer and less confusing than using the shipped `admin` database.

**Review by puregarlic — CHANGES_REQUESTED** [review]
Stellar testing job, but we should have a discussion about the things I commented on.

File: src/server/index.ts (hunk 3 above)
Comment (danvanhorn) [reply src/server/index.ts]: agreed, this is just carry over code from trying to insert the data.

File: src/dal/interface/index.ts (hunk 1 above)
Comment (danvanhorn) [reply src/dal/interface/index.ts]: I'm throwing an error now from this function. We'll probably want to handle it later up the pipeline in the request handler.

**Review by puregarlic — APPROVED** [review]
Alright, awesome. I think this is a great start to the CSE API.

**Pull request merged**